Changelog.rst

```diff
@@ -1,3 +1,13 @@
+Version 1.10.?.?
+----------------
+
+**2023-??-??**
+
+* Fixed bug that caused `cf.Data.second_element` to fail for some data
+  shapes (https://github.com/NCAS-CMS/cfdm/issues/249)
```

Comment:
In my understanding of the original issue and fix, specifically the issue occurs when the final axes has a singular size i.e. last value in the shape is `1` (for quick testing on this, I used this):

```python
def test_shape(size, shape):
    d= cfdm.Data(np.arange(size).reshape(*shape))
    return d.second_element()

# And checking, e.g:
test_shape(60, (1, 15, 2, 2))  # OK
test_shape(60, (2, 15, 2, 1))  # Errors
# Etc.
```

so perhaps we can be more specific in case users think they have to re-check work or something drastic (in the unlikely case they even read the changelog!). I'm not sure if this is the clearest wording, but something like:

```suggestion
  shapes, namely for a final axis with size one.
  (https://github.com/NCAS-CMS/cfdm/issues/249)
```